src/ray/raylet/node_manager.cc

```diff
@@ -1752,7 +1752,7 @@ void NodeManager::HandleTaskReconstruction(const TaskID &task_id) {
                "allocation via "
             << "ray.init(redis_max_memory=<max_memory_bytes>).";
         // Use a copy of the cached task spec to re-execute the task.
-        const Task task = lineage_cache_.GetTask(task_id);
+        const Task task = lineage_cache_.GetTaskOrDie(task_id);
```

Comment:
We shouldn't check fail in this function. However, the issue now is that you can't fail a task with the TaskSpec.

Is there a function from TaskID -> ReturnIDs?

Reply:
You can use `[ComputeReturnId](https://github.com/ray-project/ray/blob/master/src/ray/id.h#L68)`. Unfortunately, we can't know how many return values to put the error for without the task spec...I guess the safest option for now is to just put one return value.

Reply:
What if I returned like 10? Could that cause an issue?

Reply:
For actor tasks, the last return value is the dummy object, which isn't supposed to have any value in the object store...this could potentially break other parts of the raylet, but I'm not totally sure.